maven-core/src/main/java/org/apache/maven/project/artifact/MavenMetadataSource.java

```diff
@@ -569,6 +569,7 @@ private ProjectRelocation retrieveRelocatedProject( Artifact artifact, MetadataR
                     configuration.setValidationLevel( ModelBuildingRequest.VALIDATION_LEVEL_MINIMAL );
                     configuration.setProcessPlugins( false );
                     configuration.setRepositoryMerging( ProjectBuildingRequest.RepositoryMerging.REQUEST_DOMINANT );
+                    // TODO How to access here Maven's copy of system properties and user properties?
```

Comment:
They should be accessible through the `legacySupport` which is used just below.
Maybe something like:
```
                    MavenSession session = legacySupport.getSession();
                    if ( session != null )
                    {
                        configuration.setSystemProperties( session.getSystemProperties() );
                        configuration.setUserProperties( session.getUserProperties() );
                    }
                    else
                    {
                        configuration.setSystemProperties( getSystemProperties() );
                        configuration.setUserProperties( new Properties() );
                    }
```

Reply:
This contains a potential NPE then:
https://github.com/apache/maven/blob/cb54aa429d9e63bf78c4c808898fbef1cc01ff33/maven-core/src/main/java/org/apache/maven/project/artifact/MavenMetadataSource.java#L198-L203

Reply:
Updated spot, thanks!